### PR TITLE
plan: tiny code refine for building DataSource

### PR DESCRIPTION
**Pull request by zz-jason**
- add some comments
- only add metrics when statistics is outdated
- fix typo, change `avalableIndices` to `availableIndices`

### CODE REVIEW

**Comment by zz-jason** [comment]
@winoros @lamxTyler @zimulala PTAL

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -1579,81 +1579,99 @@ func (ds *DataSource) newExtraHandleSchemaCol() *expression.Column {
...
 		Columns:          make([]*model.ColumnInfo, 0, len(columns)),
-		availableIndices: &avalableIndices,
 	}.init(b.ctx)
-	b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SelectPriv, schemaName.L, tableInfo.Name.L, "")
 
 	var handleCol *expression.Column
 	schema := expression.NewSchema(make([]*expression.Column, 0, len(columns))...)
 	for i, col := range columns {
-		ds.Columns = append(ds.Columns, col.ToInfo())
+		ds.Columns = append(ds.Columns, columns[i].ToInfo())
```
Comment (winoros) [inline plan/logical_plan_builder.go]: why change here?

Comment (zz-jason) [reply plan/logical_plan_builder.go]: by mistake, I'll revert this change.

**Comment by winoros** [comment]
LGTM

File: statistics/table.go [hunk 2]
```diff
@@ -336,11 +336,13 @@ func (t *Table) GetRowCountByIndexRanges(sc *stmtctx.StatementContext, idxID int
 
 // PseudoTable creates a pseudo table statistics when statistic can not be found in KV store.
```
Comment (lamxTyler) [inline statistics/table.go]: Update the comment?

**Comment by coocood** [comment]
LGTM

File: plan/planbuilder.go [hunk 3]
```diff
@@ -313,62 +313,55 @@ func (b *planBuilder) detectSelectAgg(sel *ast.SelectStmt) bool {
...
 	}
-	return removeIgnores(publicIndices, ignores), true, nil
+
+	available = removeIgnoredIndex(available, ignored)
+	// If we have got "FORCE" or "USE" index hint but got no available index,
+	// we have to use table scan.
+	return &availableIndices{available, len(available) == 0}, nil
 }
 
-func removeIgnores(indices, ignores []*model.IndexInfo) []*model.IndexInfo {
```
Comment (zimulala) [inline plan/planbuilder.go]: s/removeIgnoredIndex/removeIgnoredIndices ?

File: plan/logical_plan_builder.go [hunk 4]
```diff
@@ -1579,81 +1579,99 @@ func (ds *DataSource) newExtraHandleSchemaCol() *expression.Column {
...
-				// Zero count always brings some strange problem.
-				statsTbl.Count = 100
-			}
-			metrics.PseudoEstimation.Inc()
-		}
-	}
-	indices, includeTableScan, err := availableIndices(tn.IndexHints, tableInfo)
+	b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SelectPriv, dbName.L, tableInfo.Name.L, "")
+
+	availableIdx, err := getAvailableIndices(tn.IndexHints, tableInfo)
```
Comment (zimulala) [inline plan/logical_plan_builder.go]: s/availableIdx/availableIdxs or availableIdxes ?

**Comment by zz-jason** [comment]
@zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**